src/App.tsx

```diff
@@ -1,29 +1,89 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+import { PaginationInfo } from './components/PaginationInfo';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const items: number[] = getNumbers(1, 42).map(n => n);
+
+const perPageOptions = [3, 5, 10, 20].map((el, i) => {
+  return { id: i, val: el };
+});
+
+interface PageItem {
+  id: number;
+  val: number;
+}
 
 export const App: React.FC = () => {
+  function buildPagesContainer(
+    itms: number[],
+    chunkSize: number,
+  ): PageItem[][] {
+    const contArr: PageItem[][] = [];
+    let globId = 0;
+
+    if (!chunkSize) {
+      return [];
+    }
```

Comment:
It's good practice to explicitly check for the type of `chunkSize`. For example, `if (typeof chunkSize !== 'number' || chunkSize <= 0)` would be more robust against invalid inputs.